src/index.ts

```diff
@@ -685,48 +704,36 @@ export function processProject(
   project.getSourceFiles().forEach(sourceFile => {
     const dependencies: Dependencies = new Map()
     const addDependency = createAddDependency(dependencies)
-    const functions = sourceFile
-      .getChildAtIndex(0)
-      .getChildren()
-      .reduce<string[]>((acc, child) => {
-        if (!TypeGuards.isJSDocableNode(child)) {
-          return acc
-        }
-        const typeGuardName = getTypeGuardName(child.getJsDocs())
-        if (typeGuardName === null) {
-          return acc
-        }
-        if (!TypeGuards.isExportableNode(child)) {
-          reportError(`Must be exportable:\n\n${child.getText()}\n`)
-          return acc
-        }
+
+    const functions = []
+    const exports = Array.from(sourceFile.getExportedDeclarations().values())
+    for (const exp of exports) {
+      for (const singleExport of exp) {
         if (
-          TypeGuards.isEnumDeclaration(child) ||
-          TypeGuards.isInterfaceDeclaration(child) ||
-          TypeGuards.isTypeAliasDeclaration(child)
+          Node.isTypeAliasDeclaration(singleExport) ||
+          Node.isInterfaceDeclaration(singleExport)
         ) {
-          if (!child.isExported()) {
-            reportError(`Node must be exported:\n\n${child.getText()}\n`)
```

Comment:
I kind of liked having this error report here to catch the case where you add the `ts-auto-guard:generate` and forget to `export` the type. Could you add a note to the new "Specify with annotation" section please?